doc/x-interface.md

```diff
@@ -0,0 +1,105 @@
+| `q_rs1`        | `DataWidth-1:0`            | Core > Adapter | Source register contents (rs1)                              |
+| `q_rs2`        | `DataWidth-1:0`            | Core > Adapter | Source register contents (rs2)                              |
+| `q_rs3`        | `DataWidth-1:0`            | Core > Adapter | Source register contents (rs3)                              |
+| `q_rs_valid`   | `2:0`                      | Core > Adapter | Source register valid indicator                             |
+| `q_rd_clean` | `1:0`                        | Core > Adapter | Scoreboard status of destination register. (NEW) |
+| `k_accept`     | `0:0`                      | Adapter > Core | Offload request acceptance indicator.                       |
+| `k_writeback`  | `1:0`                      | Adapter > Core | Mark outstanding accelerator writeback to`rd` (and `rd+1`). |
+
+Additionally, the handshake signals `q_ready` and `q_valid` are implemented.
+
+#### Request Transaction
+The instruction offloading process takes place according to the following scheme:
+- The offloading core asserts `q_valid` upon encountering an unknown instruction in the instruction decode stage, initiating a transaction.
+- Once `q_valid` has been asserted, `q_instr_data` must remain stable.
+- The signals `q_rs_valid[i]` indicate the status of the source registers `q_rs[i]`.
+  `q_rs_valid[i]` may initially be 0 and change to 1 during the transaction.
+  While `q_rs_valid[i]` is low, `q_rs[i]` may change.
+  Once `q_valid` and `q_rs_valid[i]` have been asserted, `q_rs_valid[i]` and `q_rs[i]` must remain stable.
+- `q_rd_clean` indicates the scoreboard entry of the destination register.
+  It may initially be 0 and change to 1 during the transaction.
+  Once `q_rd_clean` is asserted, it must remain stable.
+- The accelerator adapter asserts `q_ready`, if
+    - The instruction is accepted by any of the connected predecoders, the required source registers are marked valid by the offloading core and the destination register is clean (if writeback is expected).
+      The adapter asserts `k_accept` and `k_writeback` accordingly.
+    - The instruction is not accepted by any of the connected predecoders.
+      The adapter de-asserts `k_accept` and `k_writeback` to indicate an illegal instruction has been encountered.
+- When both `q_valid` and `q_ready` are high, the transaction is successful
+
+### Response channel
+The response channel signals are:
+
+| Signal Name   | Range                   | Description                          |
+| ------------- | ----------------------- | ------------------------------------ |
+| `p_rd`        | `4:0`                   | Destination Register Address         |
+| `p_data0`     | `DataWidth-1:0`         | Primary Writeback Data               |
+| `p_data1`     | `DataWidth-1:0`         | Secondary Writeback Data             |
+| `p_dualwb`    | `0:0`                   | Dual-Writeback Response              |
```

Comment:
Isn't that already known by the predecoder/core?

Reply:
It is known to the predecoder at the time of offloading the instruction. 
The core reserves the appropriate registers in its scoreboard according to input from the adapter (`k_writeback`) and the destination address extracted from the instruction.
But none of this is stored by either the core or the adapter. Once the request leaves the adapter, the only trace that remains is at this point in the scoreboard. Therefore the offload response must provide information as to where it writes back.